src/backend/distributed/utils/metadata_cache.c

```diff
@@ -2950,7 +2956,10 @@ ResetDistTableCacheEntry(DistTableCacheEntry *cacheEntry)
 		bool foundInCache = false;
 
 		/* delete the shard's placements */
-		pfree(placementArray);
+		if (placementArray != NULL)
+		{
+			pfree(placementArray);
```

Comment:
This check is probably not needed, since std libraries `free()` just ignores `NULL` pointers, just wanted to make things explicit.